### PR TITLE
[Test] CEPH-83573775 - Added Test Case for disconnected Installation

### PR DESCRIPTION
**Pull request by msainiRedhat**
Signed-off-by: Manisha Saini <msaini@msaini.blr.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: ceph/ceph_admin/cephadm_ansible.py [hunk 1]
```diff
@@ -27,7 +27,13 @@ def __init__(self, cluster):
             cluster: Ceph cluster object
         """
         self.cluster = cluster
-        self.admin = self.cluster.get_ceph_object("installer")
+        for host in self.cluster.get_nodes():
+            roles = host.role.role_list
+        if "network-node" in roles:
```
Comment (psathyan) [inline ceph/ceph_admin/cephadm_ansible.py:32]: The roles that are mentioned here have a one to one mapping with the daemon types listed in `ceph`. As `network-node` is not one of them, I would like us to not add this here. This breaks our design.

File: ceph/ceph_admin/cephadm_ansible.py [hunk 2]
```diff
@@ -101,6 +107,14 @@ def execute_playbook(self, playbook, extra_vars=None, extra_args=None):
         if extra_args:
             cmd += config_dict_to_string(extra_args)
 
+        # Check for Disconnected installation
+        for ceph in self.cluster.get_nodes():
+            if ceph.role == "network-node":
+                #Fetch network node hostname
+                network_node = self.cluster.get_ceph_object("network-node").node
+                network_node_hostname=network_node.hostname
+                cmd += f" -e 'custom_repo_url=http://{network_node_hostname}/repo/ceph-Tools'"
```
Comment (psathyan) [inline ceph/ceph_admin/cephadm_ansible.py:116]: If I understand correctly, in the case of disconnected install the following is true
- ceph tools repo is internally hosted
- container image registry is internally hosted

If correct, then we don't need to add any code changes and the testing can be done via the CLI options

`--add-repo` can be used to the tools repo

In `.cephci.yaml`, you would need to modify the `registry_credentials` section to point to the right one.

File: conf/pacific/cephadm/disconnected_installation.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (psathyan) [inline conf/pacific/cephadm/disconnected_installation.yaml:1]: Going down the path of not having `network-node`... I believe we can avoid this configuration and reuse the existing ones.

**Review by psathyan — CHANGES_REQUESTED** [review]
Thanks for raising this PR.

Can we connect offline as I would like to review why the design and requirements?

**Comment by psathyan** [comment]
@msainiRedhat Would you like to continue using this PR or would it make sense to create another one for the required changes?

**Comment by msainiRedhat** [comment]
@psathyan I will create another one for the required changes

**Comment by psathyan** [comment]
> @psathyan I will create another one for the required changes

@msainiRedhat Please go ahead close this one.

**Comment by vdas-redhat** [comment]
@msainiRedhat Can we close this PR ?

**Comment by msainiRedhat** [comment]
as per latest discussion, there are no changes required for disconnected installation. Closing this pull request as these changes are not needed

**Pull request closed without merging**